src/main/java/com/faforever/client/replay/LiveReplayService.java

```diff
@@ -0,0 +1,124 @@
+package com.faforever.client.replay;
+
+import com.faforever.client.config.ClientProperties;
+import com.faforever.client.domain.GameBean;
+import com.faforever.client.fx.JavaFxUtil;
+import com.faforever.client.game.GameService;
+import com.faforever.client.i18n.I18n;
+import com.faforever.client.notification.NotificationService;
+import com.faforever.client.notification.TransientNotification;
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.util.Pair;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.stereotype.Service;
+import org.springframework.util.Assert;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.util.concurrent.Future;
+
+import static com.faforever.client.util.Assert.checkNullIllegalState;
+
+@Lazy
+@Service
+@RequiredArgsConstructor
+public class LiveReplayService implements InitializingBean, DisposableBean {
+
+  private final ClientProperties clientProperties;
+  private final TaskScheduler taskScheduler;
+  private final NotificationService notificationService;
+  private final I18n i18n;
+  private final ReplayService replayService;
+  private final GameService gameService;
+
+  private Integer watchDelaySeconds;
+  private Future<?> futureTask;
+  private final ObjectProperty<Pair<Integer, LiveReplayAction>> trackingReplayProperty = new SimpleObjectProperty<>(null);
+
+  @Override
+  public void afterPropertiesSet() throws Exception {
+    watchDelaySeconds = clientProperties.getReplay().getWatchDelaySeconds();
+    JavaFxUtil.addListener(gameService.gameRunningProperty(), (observable, oldValue, newValue) -> {
+      if (newValue.equals(true) && trackingReplayProperty.getValue() != null) {
+        stopTrackingReplay();
+      }
+    });
```

Comment:
If we aren't using the old value I would just make this an invalidation listener and retrieve the new value
```suggestion
    JavaFxUtil.addListener(gameService.gameRunningProperty(), observable -> {
      if (trackingReplayProperty.getValue() != null && gameService.isGameRunning()) {
        stopTrackingReplay();
      }
    });
```